### ISSUE TITLE
Clicking on Settings Tab throws "500 Internal Server Error"(Zikula 2.0.14) 

### ISSUE THREAD

**Issue by prbt2016** [zikula/core#3983]
Summary : After successful manual installation of Zikula 2.0.14. Clicking on Settings tab throws Oops! An Error Occurred
The server returned a "500 Internal Server Error".

| Q                 | A
| ----------------- | ---
| Zikula Version    | 2.0.14
| PHP Version       | 5.5.38

## Expected Behavior
Settings tab should be loaded properly.

## Current Behavior
Settings tab is not loaded properly . It is broken.

![test](https://user-images.githubusercontent.com/17871448/68851962-e616ca80-06fc-11ea-98d6-72c0954b41b6.png)

Error Log thrown in /var/logs/prod.log shows : 
request.CRITICAL: Uncaught PHP Exception Symfony\Component\Form\Exception\TransformationFailedException: "Unable to transform value for property path "[UseCompression]": Expected a Boolean." at /home/soft/public_html/manual/zikula2014/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php line 1087 {"exception":"[object] (Symfony\\Component\\Form\\Exception\\TransformationFailedException(code: 0): Unable to transform value for property path \"[UseCompression]\": Expected a Boolean. at /home/soft/public_html/manual/zikula2014/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php:1087, Symfony\\Component\\Form\\Exception\\TransformationFailedException(code: 0): Expected a Boolean. at /home/soft/public_html/manual/zikula2014/vendor/symfony/symfony/src/Symfony/Component/Form/Extension/Core/DataTransformer/BooleanToStringTransformer.php:51)"} []

## Steps to Reproduce

1.Manually Install Zikula 2.0.14
2.Go to Administration panel -> Settings and '500 Internal Server Error' is thrown.

## Context
This issue affects fresh installation of Zikula 2.014 and also it affects upgrade (Zikula 2.0.13 ->Zikula 2.0.14). In both cases 'Settings' tab is broken. 

Note : I have zlib output compression(zlib.output_compression=On) enabled in php.ini file. This issue isn't present in fresh install of Zikula 2.0.13.

**Comment by Guite** [comment]
I wasn't able to reproduce this behaviour yet.
Can you please try the following patch?

Replace https://github.com/zikula/core/blob/master/src/system/SettingsModule/Controller/SettingsController.php#L67-L73 by:
```
        $variables = $variableApi->getAll(VariableApi::CONFIG);
        $variables['UseCompression'] = (bool)$variables['UseCompression'];
        $form = $this->createForm(MainSettingsType::class,
            $variables, [
                'languages' => $installedLanguageNames,
                'profileModules' => $this->formatModuleArrayForSelect($extensionRepository, $profileModules),
                'messageModules' => $this->formatModuleArrayForSelect($extensionRepository, $messageModules)
            ]
        );
```

I am pretty sure this fixes the problem, but it would be good to have it confirmed.

**Comment by Guite** [comment]
I wasn't able to reproduce this behaviour yet.
Can you please try the following patch?

Replace https://github.com/zikula/core/blob/master/src/system/SettingsModule/Controller/SettingsController.php#L67-L73 by:
```
        $variables = $this->getSystemVars();
        $variables['UseCompression'] = (bool)$variables['UseCompression'];
        $form = $this->createForm(MainSettingsType::class,
            $variables,
            [
                'translator' => $this->get('translator.default'),
                'languages' => $installedLanguageNames,
                'profileModules' => $this->formatModuleArrayForSelect($profileModules),
                'messageModules' => $this->formatModuleArrayForSelect($messageModules)
            ]
        );
```

I am pretty sure this fixes the problem, but it would be good to have it confirmed.

**Comment by Guite** [comment]
Sorry I used the wrong branch in my initial response. Please try the latest version I've posted above.

**Comment by prbt2016** [comment]
Hello @Guite ;

> Sorry I used the wrong branch in my initial response. Please try the latest version I've posted above.

Code does work fine but it breaks the Settings->Security->Security Center->HTMLPurifier settings module. Clicking on the HTMLPurifier settings now throws an Internal server error. 

![test](https://user-images.githubusercontent.com/17871448/69032540-bfa9a580-0a02-11ea-91c1-c7d0f4c492b7.png)

This is the error log thrown in /var/logs/prod.log

[2019-11-18 12:52:51] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Undefined class constant 'STRING'" at /{{PATH}}/{{TO}}/{{ZIKULA}}/system/SecurityCenterModule/Controller/ConfigController.php line 453 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Undefined class constant 'STRING' at /{{PATH}}/{{TO}}/{{ZIKULA}}/system/SecurityCenterModule/Controller/ConfigController.php:453)"} []

Could you please test that the above change doesn't break any other module or setting?

**Comment by Guite** [comment]
This is not related, but caused by a vendor update.
Will fix both.

Thanks for reporting!

**Comment by Guite** [comment]
Both fixed in 14e4d197d16a3abb39a1d4312497628627f7b300
